### PR TITLE
[Mod] Account for duplicated mentions

### PR DESCRIPTION
**Pull request by SharkyTheKing**
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes
Allow for mention spam to account for duplicated mentions.
Added `modset mentionspam strict` bool to toggle the system to account for duplicates or only uniques.
Added settings information in showsettings.
Fixed wording when you adjust warn/kick/ban settings (removed `different`)

### CODE REVIEW

File: redbot/cogs/mod/settings.py [hunk 1]
```diff
@@ -106,6 +112,24 @@ async def mentionspam(self, ctx: commands.Context):
...
+    @mentionspam.command(name="strict")
+    @commands.guild_only()
+    async def mentionspam_strict(self, ctx: commands.Context, toggle: bool):
+        """
+        Sets settings to allow for duplicate or only different mentions.
+
+        If enabled all mentions will count including duplicated mentions.
+        If disabled only unique mentions will count.
+
+        Default is settings is False.
```
Comment (ltzmax) [inline redbot/cogs/mod/settings.py]: this would look better `Default setting is False.` instead of `Default is settings is False.`

**Review by ltzmax — CHANGES_REQUESTED** [review]

File: redbot/cogs/mod/settings.py (hunk 1 above)
Comment (fixator10) [reply redbot/cogs/mod/settings.py]:
```suggestion
        Default setting is False.
```

Comment (SharkyTheKing) [reply redbot/cogs/mod/settings.py]: Wording changed, ty.

**Review by ltzmax — DISMISSED** [review]
looks good to me. I have done testing and no issues was found.

File: redbot/cogs/mod/settings.py [hunk 2]
```diff
@@ -54,6 +55,11 @@ async def modset_showsettings(self, ctx: commands.Context):
             if ban_mention_spam
             else _("No")
         )
+        msg += (
+            _("Mention Spam Strict: All mentions will count including duplicates\n")
+            if strict_mention_spam
+            else _("Mention Spam Strict: Only unique mention will count\n")
```
Comment (Flame442) [inline redbot/cogs/mod/settings.py]:
```suggestion
            else _("Mention Spam Strict: Only unique mentions will count\n")
```

File: redbot/cogs/mod/settings.py [hunk 3]
```diff
@@ -106,6 +112,24 @@ async def mentionspam(self, ctx: commands.Context):
         Manage the automoderation settings for mentionspam.
         """
 
+    @mentionspam.command(name="strict")
+    @commands.guild_only()
+    async def mentionspam_strict(self, ctx: commands.Context, toggle: bool):
+        """
+        Sets settings to allow for duplicate or only different mentions.
```
Comment (Flame442) [inline redbot/cogs/mod/settings.py]: Please reword this short-doc to be clearer.

File: redbot/cogs/mod/settings.py [hunk 4]
```diff
@@ -106,6 +112,24 @@ async def mentionspam(self, ctx: commands.Context):
         Manage the automoderation settings for mentionspam.
         """
 
+    @mentionspam.command(name="strict")
+    @commands.guild_only()
+    async def mentionspam_strict(self, ctx: commands.Context, toggle: bool):
```
Comment (Flame442) [inline redbot/cogs/mod/settings.py]: The other commands in this cog are toggles, not passed booleans. You should follow that pattern.

File: redbot/cogs/mod/settings.py [hunk 5]
```diff
@@ -106,6 +112,24 @@ async def mentionspam(self, ctx: commands.Context):
...
+        Sets settings to allow for duplicate or only different mentions.
+
+        If enabled all mentions will count including duplicated mentions.
+        If disabled only unique mentions will count.
+
+        Default setting is False.
+        """
+        if toggle is True:
+            await self.config.guild(ctx.guild).mention_spam.strict.set(True)
+            return await ctx.send(_("Mention spam will now account for same mentions."))
```
Comment (Flame442) [inline redbot/cogs/mod/settings.py]:
```suggestion
            return await ctx.send(_("Mention spam will now account for multiple mentions of the same user."))
```

File: redbot/cogs/mod/settings.py [hunk 6]
```diff
@@ -106,6 +112,24 @@ async def mentionspam(self, ctx: commands.Context):
...
+        elif toggle is False:
+            await self.config.guild(ctx.guild).mention_spam.strict.set(False)
+            return await ctx.send(_("Mention spam will only account for different mentions."))
```
Comment (Flame442) [inline redbot/cogs/mod/settings.py]:
```suggestion
            return await ctx.send(_("Mention spam will only account for mentions of different users."))
```

**Review by Flame442 — CHANGES_REQUESTED** [review]
The code itself works fine, just some UX issues that need tackling.

File: redbot/cogs/mod/settings.py (hunk 3 above)
Comment (SharkyTheKing) [reply redbot/cogs/mod/settings.py]: Reworded, not sure if the new one is better. Will wait for you to check.

**Review by Flame442 — DISMISSED** [review]

**Review by Flame442 — APPROVED** [review]

**Pull request merged**